specreduce/tracing.py

```diff
@@ -110,3 +113,164 @@ def __post_init__(self):
+    guess: float = None
+    window: int = None
+    disp_axis = 1
+    
+    def __post_init__(self,
+                      # Saxis=0, Waxis=1, display=False
+                      ):
+        if self.bins < 4:
+            raise ValueError('bins must be >= 4')
+        if not isinstance(self.bins, int):
+            warnings.warn('TRACE: Converting bins to int')
+            self.bins = int(self.bins)
+
+        if (self.window is not None
+            and (self.window > self.image.shape[self.disp_axis]
+                 or self.window < 1)):
+            raise ValueError("window must be >= 2 and less than the length of "
+                             "the image's spatial direction")
+        elif self.window is not None and not isinstance(self.window, int):
+            warnings.warn('TRACE: Converting window to int')
+            self.window = int(self.window)
+
+        # set max peak location by user choice or wavelength with max avg flux
+        ztot = np.nansum(self.image, axis=1) / self.image.shape[1]
+        peak_y = self.guess if self.guess is not None else np.nanargmax(ztot)
+        # (peak finder can be bad if multiple objects are on slit...)
+
+        # guess the peak width as the FWHM, roughly converted to gaussian sigma
+        yy = np.arange(len(ztot))
+        yy_above_half_max = np.sum(ztot > (np.nanmax(ztot) / 2))
+        width_guess = yy_above_half_max / 2.355
+
+        # enforce some (maybe sensible?) rules about trace peak width
+        width_guess = (2 if width_guess < 2
+                       else 25 if width_guess > 25
+                       else width_guess)
+
+        # fit a Gaussian to peak for fall-back answer, but don't use yet
+        g1d_init = models.Gaussian1D(amplitude=np.nanmax(ztot),
+                                     mean=peak_y, stddev=width_guess)
+        offset_init = models.Const1D(np.nanmedian(ztot))
+        profile = g1d_init + offset_init
+
+        fitter = fitting.LevMarLSQFitter()
+        popt_tot = fitter(profile, yy, ztot)
+
+        # restrict fit to window (if one exists)
+        ilum2 = (yy if self.window is None
+                 else yy[np.arange(peak_y - self.window,
+                                   peak_y + self.window, dtype=int)])
+
+        x_bins = np.linspace(0, self.image.shape[self.disp_axis],
+                             self.bins + 1, dtype=int)
+        y_bins = np.tile(np.nan, self.bins)
+
+        for i in range(self.bins):
+            # repeat earlier steps to create gaussian fit for each bin
+            z_i = np.nansum(self.image[ilum2, x_bins[i]:x_bins[i+1]],
+                            axis=self.disp_axis)
+            peak_y_i = ilum2[np.nanargmax(z_i)]
+
+            yy_i_above_half_max = np.sum(z_i > (np.nanmax(z_i) / 2))
+            width_guess_i = yy_i_above_half_max / 2.355
+            width_guess_i = (2 if width_guess_i < 2
+                             else 25 if width_guess_i > 25
+                             else width_guess_i)
```

Comment:
See above
> While I'm all for intelligent software, I share @kecnry's concern about these hardcoded values. I'm wondering if these values can be extracted from some attribute of the image that's coming in? But that's outside my expertise

Reply:
should use `astropy.stats.gaussian_fwhm_to_sigma` here in lieu of 2.355 for both precision and clarity.